### PR TITLE
Updated/Fixed the code coverage pipeline

### PR DESCRIPTION
**Pull request by igomez06**
## Issue: 
 
https://github.com/rancher/qa-tasks/issues/775
## Problem
 
 The code coverage Jenkins pipeline was no longer working consistently and would fail during setup most of the time.
## Solution
 Updated the Jenkinsfile, Dockerfile, as well as other aspects of the pipeline so the code coverage job would work more consistently. Also made a fix to the charts version integration test that was leaving rancher in a weird state. Added v2prov integration tests to the pipeline as well
 
## Testing

 Jenkins pipeline is working consistently

### CODE REVIEW

File: tests/v2/codecoverage/package/Dockerfile [hunk 1]
```diff
@@ -156,13 +158,13 @@ RUN ln -s /usr/bin/cni /usr/bin/bridge && \
...
     mkdir -p /var/lib/rancher/k3s/agent/images/ && \
-    curl -sfL ${ETCD_URL} | tar xvzf - --strip-components=1 -C /usr/bin/ etcd-${CATTLE_ETCD_VERSION}-linux-${ARCH}/etcdctl && \
+    curl -sfL ${ETCD_URL} | tar xvzf - --strip-components=1 --no-same-owner -C /usr/bin/ etcd-${CATTLE_ETCD_VERSION}-linux-${ARCH}/etcdctl && \
     curl -sLf https://github.com/rancher/telemetry/releases/download/${TELEMETRY_VERSION}/telemetry-${ARCH} > /usr/bin/telemetry && \
     chmod +x /usr/bin/tini /usr/bin/telemetry && \
     mkdir -p /var/lib/rancher-data/driver-metadata
 
-ENV CATTLE_UI_VERSION 2.7.2-rc8
-ENV CATTLE_DASHBOARD_UI_VERSION v2.7.2-rc8
+ENV CATTLE_UI_VERSION 2.7.5-rc2
```
Comment (markusewalker) [inline tests/v2/codecoverage/package/Dockerfile:166]: Should we update this to 2.7.5-rc3 as that's the final RC?

Comment (igomez06) [reply tests/v2/codecoverage/package/Dockerfile:166]: Well that's what in the release/v2.7 main branch. I update it according to that docker file.

Comment (igomez06) [reply tests/v2/codecoverage/package/Dockerfile:166]: Maybe should another update once we release v2.7.5

Comment (markusewalker) [reply tests/v2/codecoverage/package/Dockerfile:166]: Whichever makes more sense for the code coverage, that just caught my eye during a review. Everything else lgtm

**Review by caliskanugur — COMMENTED** [review]
I've got one nit, looks really good

**Review by izaac — DISMISSED** [review]
LGTM

**Review by markusewalker — APPROVED** [review]

**Review by maxsokolovsky — DISMISSED** [review]

File: tests/v2/codecoverage/package/Dockerfile (hunk 1 above)
Comment (anupama2501) [reply tests/v2/codecoverage/package/Dockerfile:166]: Do we want to update this as and when a new rancher version releases? not that it is essential but thought would ask.

**Review by anupama2501 — APPROVED** [review]
had a question/needed a confirmation. LGTM  otherwise

**Review by slickwarren — DISMISSED** [review]
lgtm

File: tests/v2/codecoverage/package/Dockerfile (hunk 1 above)
Comment (igomez06) [reply tests/v2/codecoverage/package/Dockerfile:166]: Yes I was planning on updating it after the new rancher releases. What do you all think?

File: tests/v2/integration/catalogv2/system_charts_version_test.go [hunk 2]
```diff
@@ -72,6 +80,47 @@ func (w *SystemChartsVersionSuite) resetSettings() {
...
+		} else if err != nil {
+			return false, err
+		}
+
+		newDeployment := &appv1.Deployment{}
+		err = scheme.Scheme.Convert(deployment, newDeployment, deployment.GroupVersionKind())
+		if err != nil {
+			return false, err
+		}
+		if newDeployment.Status.ReadyReplicas == int32(1) {
```
[the remark this answers is not in the document]
Comment (igomez06) [reply tests/v2/integration/catalogv2/system_charts_version_test.go]: changed

**Review by markusewalker — APPROVED** [review]

File: tests/v2/codecoverage/package/Dockerfile (hunk 1 above)
Comment (slickwarren) [reply tests/v2/codecoverage/package/Dockerfile:166]: sgtm, we had a similar process for the old ontag jobs

**Review by slickwarren — DISMISSED** [review]

**Review by markusewalker — APPROVED** [review]

**Review by markusewalker — APPROVED** [review]

**Pull request merged**